File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/uncomment/direct/Java/Python/s527301188_commented.py
import sys

def main():
    tokens = sys.stdin.readline().split()
    stack = []
    for token in tokens:
        c = token[0]
        if c in {'+', '-', '*'}:
            b = stack.pop()
            a = stack.pop()
            if c == '+':
                stack.append(a + b)
            elif c == '-':
                stack.append(a - b)
            else:
                stack.append(a * b)
        else:
            stack.append(int(token))
    print(stack[0])

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/javadoc/uncomment/direct/Java/Go/s527301188_commented.go
package main

import (
	"bufio"
	"os"
	"strconv"
	"strings"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	line, _ := reader.ReadString('\n')
	strs := strings.Fields(line)

	var stack []int
	for _, str := range strs {
		switch str[0] {
		case '+', '-', '*':
			b := stack[len(stack)-1]
			stack = stack[:len(stack)-1]
			a := stack[len(stack)-1]
			stack = stack[:len(stack)-1]
			var result int
			switch str[蕹
			case '+':
				result = a + b
			case '-':
				result = a - b
			case '*':
				result = a * b
			}
			stack = append(stack, result)
		default:
			num, _ := strconv.Atoi(str)
			stack = append(stack, num)
		}
	}
	println(stack[0])
}

// End of Code
